# HG changeset patch
# User David Grundberg <davidg@cs.umu.se>
# Date 1257788544 18000
#      Mon Nov 09 12:42:24 2009 -0500
# Node ID cc5bb7b5679fe3fd83cfe93c7fd27d8c55390a03
# Parent  2c279308f6abf156b99cda7ae413089ffcc6fc99
use AMD_CPPFLAGS when checking for UMFPACK

diff --git a/ChangeLog b/ChangeLog
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,8 +1,12 @@
+2009-11-09  David Grundberg  <davidg@cs.umu.se>
+
+	* configure.ac: Use AMD_CPPFLAGS while checking for UMFPACK.
+
 2009-10-29  Thomas Treichl  <Thomas.Treichl@gmx.net>
 
 	* configure.in: Add support for dlopen on Mac systems.
 
 2009-10-20  Jaroslav Hajek  <highegg@gmail.com>
 
 	* NEWS: Update.
 
diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -971,22 +971,25 @@ OCTAVE_CHECK_LIBRARY(cxsparse, CXSparse,
   [CXSparse library not found.  This will result in some lack of functionality for sparse matrices.],
   [suitesparse/cs.h ufsparse/cs.h amd/cs.h cs.h],
   [cs_di_sqr],
   [C++], [don't use CXSparse library, disable some sparse functionality])
 
 # Check for UMFPACK library.
 
 save_LIBS="$LIBS"
+save_CPPFLAGS="$CPPFLAGS"
 LIBS="$AMD_LDFLAGS $AMD_LIBS $BLAS_LIBS $FLIBS $LIBS"
+CPPFLAGS="$AMD_CPPFLAGS $CPPFLAGS"
 OCTAVE_CHECK_LIBRARY(umfpack, UMFPACK,
   [UMFPACK not found.  This will result in some lack of functionality for sparse matrices.],
   [suitesparse/umfpack.h ufsparse/umfpack.h umfpack/umfpack.h umfpack.h],
   [umfpack_zi_get_determinant],
   [], [don't use UMFPACK, disable some sparse functionality])
+CPPFLAGS="$save_CPPFLAGS"
 LIBS="$save_LIBS"
 
 if test -z "$UMFPACK_LIBS"; then
   ## Invalidate the cache and try again with -lcblas.
   $as_unset ac_cv_lib_umfpack_umfpack_zi_get_determinant
   save_LIBS="$LIBS"
   LIBS="-lcblas $AMD_LDFLAGS $AMD_LIBS $BLAS_LIBS $FLIBS $LIBS"
   OCTAVE_CHECK_LIBRARY(umfpack, UMFPACK,
